Templates/AppSource App/.github/workflows/CICD.yaml

```diff
@@ -66,7 +65,25 @@ jobs:
         uses: microsoft/AL-Go-Actions/ReadSettings@main
         with:
           shell: powershell
-          get: type, powerPlatformSolutionFolder
+          get: type,powerPlatformSolutionFolder,useSubModules
+
+      - name: Read subModules token
+        id: ReadSubModulesToken
+        if: env.useSubModules != 'false' && env.useSubModules != ''
+        uses: microsoft/AL-Go-Actions/ReadSecrets@main
+        with:
+          shell: ${{ inputs.shell }}
+          gitHubSecrets: ${{ toJson(secrets) }}
+          getSecrets: '-subModulesToken'
+
+      - name: Checkout Submodules
+        if: env.useSubModules != 'false' && env.useSubModules != ''
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        with:
+          ref: ${{ inputs.checkoutRef }}
+          lfs: true
+          submodules: ${{ env.useSubModules }}
+          token: '${{ fromJson(steps.ReadSubModulesToken.outputs.Secrets).subModulesToken }}'
```

Comment:
How do you handle the case where the git submodule is a public repo and no extra token is needed?

Reply:
It falls back to using github_token as far as I can see.